app/scripts/background.js

```diff
@@ -685,6 +689,23 @@ export function setupController(
     }
   };
 
+  /**
+   *
+   * @param {string} connectionId - ID used to identify the connection in openMetamaskConnections
+   */
+  onDisconnectRemote = (connectionId) => {
```

Comment:
The `endOfStream` callbacks were intended to capture this same scenario (or so I thought). Are we now decrementing the number of open connections twice?

Reply:
Ah, great point.

In that case, are we decrementing twice for those three types of environments specifically?